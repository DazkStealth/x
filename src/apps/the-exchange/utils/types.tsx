import { BridgingProvider, ExchangeOffer } from '@etherspot/prime-sdk/dist/sdk/data';
import { Route } from '@lifi/sdk';

export enum CardPosition {
    SWAP = 'SWAP',
    RECEIVE = 'RECEIVE'
}

export type SwapType = {
    fromAmount: number,
    fromTokenAddress: string,
    fromChainId: number;
    fromTokenDecimals: number;
    toTokenAddress: string,
    toChainId: number,
    toTokenDecimals: number;
    slippage: number;
    fromAccountAddress?: string
    provider?: BridgingProvider;
}

export interface SwapOffer {
    tokenAmountToReceive: number;
    offer: Route | ExchangeOffer;
}
