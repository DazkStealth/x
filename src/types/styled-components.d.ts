import 'styled-components';

// theme
import { Theme } from '../theme';

declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}
