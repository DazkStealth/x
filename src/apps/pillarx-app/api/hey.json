{
    "projection": [
      {
        "meta": {},
        "data": {
          "total_wallet_balance": 3601.2644252510418,
          "wallet": "0x11041744893fa72629ab93ea8adaf35a1dc24aa5",
          "wallets": [
            "0x11041744893fa72629ab93ea8adaf35a1dc24aa5"
          ],
          "total_realized_pnl": 1657.3989,
          "total_unrealized_pnl": 683.2397000000001,
          "assets": [
            {
              "asset": {
                "name": "Arbitrum",
                "symbol": "ARB",
                "id": 511,
                "contracts": [
                  "0x912ce59144191c1204e64559fe8253a0e49e6548",
                  "0xb50721bcf8d664c30412cfbc6cf7a15145234ad1",
                  "0xf823c3cd3cebe0a1fa952ba88dc9eef8e0bf46ad"
                ],
                "logo": "https://metacore.mobula.io/55419fdf33d8a53e61b0bd55aa6a94a855c34a48cb62ca4766e5ac89a4e90887.png",
                "blockchains": [
                  "Arbitrum",
                  "Ethereum",
                  "Arbitrum Nova"
                ]
              },
              "realized_pnl": 102.3196,
              "unrealized_pnl": -95.9317,
              "allocation": 19.5,
              "price": 1.194947271373868,
              "price_bought": 0,
              "price_change_24h": -3.241148,
              "price_change_1h": 0,
              "total_invested": 0,
              "min_buy_price": 1.3450628571428571,
              "max_buy_price": 1.5500332634003993,
              "estimated_balance": 702.2078712586506,
              "token_balance": 587.64757917,
              "cross_chain_balances": {
                "Arbitrum": {
                  "balance": 587.64757917,
                  "balanceRaw": "587647579169999945728",
                  "chainId": "42161",
                  "address": "0x912ce59144191c1204e64559fe8253a0e49e6548"
                }
              },
              "contracts_balances": [
                {
                  "balance": 587.64757917,
                  "balanceRaw": "587647579169999945728",
                  "chainId": "42161",
                  "address": "0x912ce59144191c1204e64559fe8253a0e49e6548",
                  "decimals": 18
                }
              ]
            },
      
            {
              "asset": {
                "name": "Orbit Protocol",
                "symbol": "ORBIT",
                "id": 102480544,
                "contracts": [
                  "0x42e12d42b3d6c4a74a88a61063856756ea2db357"
                ],
                "logo": "https://metacore.mobula.io/958c9264cc15c3fdf2d6ca544010be9d8385d373d7e441c6a9239e07af301e8c.png",
                "blockchains": [
                  "Blast"
                ]
              },
              "realized_pnl": 0,
              "unrealized_pnl": -29.3324,
              "allocation": 0.15,
              "price": 0.1383163581556911,
              "price_bought": 0,
              "price_change_24h": -14.07113,
              "price_change_1h": 0,
              "total_invested": 0,
              "min_buy_price": 0.8745584448725503,
              "max_buy_price": 0.8745584448725503,
              "estimated_balance": 5.510617266881962,
              "token_balance": 39.8406764056,
              "cross_chain_balances": {
                "Blast": {
                  "balance": 39.84067640565484,
                  "balanceRaw": "39840676405654839296",
                  "chainId": "81457",
                  "address": "0x42e12d42b3d6c4a74a88a61063856756ea2db357"
                }
              },
              "contracts_balances": [
                {
                  "balance": 39.84067640565484,
                  "balanceRaw": "39840676405654839296",
                  "chainId": "81457",
                  "address": "0x42e12d42b3d6c4a74a88a61063856756ea2db357",
                  "decimals": 18
                }
              ]
            }
          ],
          "total_pnl_history": {
            "24h": {
              "realized": 0,
              "unrealized": -114.0661999999999
            },
            "7d": {
              "realized": 0,
              "unrealized": -169.20839999999998
            },
            "30d": {
              "realized": 74.44840000000019,
              "unrealized": 91.57700000000004
            },
            "1y": {
              "realized": 350.1728000000002,
              "unrealized": 955.3388000000001
            }
          }
        },
        "layout": "LIST_HORIZONTAL",
        "id": "walletPortfolio"
      }
    ]
  }