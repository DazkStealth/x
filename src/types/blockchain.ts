import { BigNumberish } from 'ethers';

export interface Transaction {
  id: string;
  value: BigNumberish;
  to: string;
  data?: string;
  status: string;
  hash?: string;
  userOpHash?: string;
}
