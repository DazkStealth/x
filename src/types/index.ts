export interface AppManifest {
  title: string;
  description: string;
  translations: RecordPerKey<string>;
}

export interface RecordPerKey<T> {
  [key: string]: T;
}
