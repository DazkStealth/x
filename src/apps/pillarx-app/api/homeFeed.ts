/* eslint-disable no-console */
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { addMiddleware } from '../../../store';
import { ApiResponse } from '../../../types/api';

export const homeFeedApi = createApi({
    reducerPath: 'homeFeedApi',
    baseQuery: fetchBaseQuery({ baseUrl: 'https://feed-nubpgwxpiq-uc.a.run.app/' }),
    endpoints: (builder) => ({
        getTilesInfo: builder.query<ApiResponse, { page: number; address: string }>({
            query: ({ page, address }) => `?page=${page}&address=${address}`,
        }),
    })
})

/**
 * Add this to the store
 */
addMiddleware(homeFeedApi);

export const { useGetTilesInfoQuery } = homeFeedApi;
