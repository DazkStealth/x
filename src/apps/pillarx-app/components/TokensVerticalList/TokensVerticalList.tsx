// types
import { TokenData } from '../../../../types/api';

// components
import HorizontalToken from '../HorizontalToken/HorizontalToken';
import Body from '../Typography/Body';

type TokensVerticalListProps = {
    position: 'left' | 'right';
    data: TokenData[];
    title?: string;
};

const TokensVerticalList = ({ position, data, title }: TokensVerticalListProps ) => {
    const topThreeTokens = data.slice(0, 3);

    return(
        <div className={`flex flex-col flex-1 mobile:p-0 mobile:border-0 ${position === 'left' ? 'pr-10 border-r-[3px] border-[#1F1D23]' : 'pl-10'}`}>
            {title && <Body className='text-purple_light'>{title}</Body>}
            <div className="flex flex-col">
                {topThreeTokens.map((token, index) => 
                    <HorizontalToken key={index} tokenIndex={index + 1} tokenName={token.name} tokenSymbol={token.symbol} tokenValue={undefined} percentage={undefined}/>
                )}
            </div>
        </div>
    );
};

export default TokensVerticalList;
