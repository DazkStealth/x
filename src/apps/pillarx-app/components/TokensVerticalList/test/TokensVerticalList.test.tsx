import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

// components
import TokensVerticalList from '../TokensVerticalList';

// types
import { TokenData } from '../../../../../types/api';

jest.mock('../../HorizontalToken/HorizontalToken', () => 'HorizontalToken');
jest.mock('../../Typography/Body', () => 'Body');

describe('<TokensVerticalList />', () => {
  const mockData: TokenData[] = [
    { 
      id: 1,
      name: 'Token A', 
      symbol: 'TKA',
      contracts: [{ address: '0x1', blockchain: 'Ethereum' }],
      logo: 'logoA.png',
      trending_score: 90,
      platforms: [{ name: 'Platform A', rank: 1, weight: 0.5 }],
      price_change_24h: 5,
      pair: 'TKA/USD'
    },
    { 
      id: 2,
      name: 'Token B', 
      symbol: 'TKB',
      contracts: [{ address: '0x2', blockchain: 'Ethereum' }],
      logo: 'logoB.png',
      trending_score: 80,
      platforms: [{ name: 'Platform B', rank: 2, weight: 0.4 }],
      price_change_24h: -3,
      pair: 'TKB/USD'
    },
  ];

  it('renders correctly and matches snapshot for left position', () => {
    const tree = renderer
      .create(<TokensVerticalList position="left" data={mockData} title='tile left' />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly and matches snapshot for right position', () => {
    const tree = renderer
      .create(<TokensVerticalList position="right" data={mockData} title='title right' />)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct title', () => {
    const tree = renderer.create(<TokensVerticalList position='left' data={mockData} title='title left' />).toJSON() as ReactTestRendererJSON;
    const titleProp = tree.children?.[0] as ReactTestRendererJSON;

    expect(titleProp).not.toBeNull();
    expect(titleProp.type).toBe('Body');
    expect(titleProp.children).toContain('title left');
    expect(tree.children?.length).toBe(2);
  });

  it('renders correctly without title when title is undefined', () => {
    const tree = renderer.create(<TokensVerticalList position='left' data={mockData} />).toJSON() as ReactTestRendererJSON;
    
    expect(tree.children?.length).toBe(1);
  });

  it('renders the correct number of HorizontalToken components', () => {
    const tree = renderer.create(<TokensVerticalList position='left' data={mockData} title='title left' />).toJSON() as ReactTestRendererJSON;
    const tokensList = tree.children?.[1] as ReactTestRendererJSON;
    const horizontalTokens = tokensList.children as ReactTestRendererJSON[];
    
    expect(horizontalTokens.length).toBe(mockData.length);
  });

  it('passes the correct props to HorizontalToken components', () => {
    const tree = renderer.create(<TokensVerticalList position='left' data={mockData} title='title left' />).toJSON() as ReactTestRendererJSON;
    const tokensList = tree.children?.[1] as ReactTestRendererJSON;
    const horizontalTokens = tokensList.children as ReactTestRendererJSON[];
    
    horizontalTokens.forEach((tokenElement, index) => {
      expect(tokenElement.props.tokenIndex).toBe(index + 1);
      expect(tokenElement.props.tokenName).toBe(mockData[index].name);
      expect(tokenElement.props.tokenSymbol).toBe(mockData[index].symbol);
      expect(tokenElement.props.tokenValue).toBeUndefined();
      expect(tokenElement.props.percentage).toBeUndefined();
    });
  });
});
