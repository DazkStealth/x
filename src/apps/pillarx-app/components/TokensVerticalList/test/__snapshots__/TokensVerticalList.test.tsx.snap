// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TokensVerticalList /> renders correctly and matches snapshot for left position 1`] = `
<div
  className="flex flex-col flex-1 mobile:p-0 mobile:border-0 pr-10 border-r-[3px] border-[#1F1D23]"
>
  <Body
    className="text-purple_light"
  >
    tile left
  </Body>
  <div
    className="flex flex-col"
  >
    <HorizontalToken
      tokenIndex={1}
      tokenName="Token A"
      tokenSymbol="TKA"
    />
    <HorizontalToken
      tokenIndex={2}
      tokenName="Token B"
      tokenSymbol="TKB"
    />
  </div>
</div>
`;

exports[`<TokensVerticalList /> renders correctly and matches snapshot for right position 1`] = `
<div
  className="flex flex-col flex-1 mobile:p-0 mobile:border-0 pl-10"
>
  <Body
    className="text-purple_light"
  >
    title right
  </Body>
  <div
    className="flex flex-col"
  >
    <HorizontalToken
      tokenIndex={1}
      tokenName="Token A"
      tokenSymbol="TKA"
    />
    <HorizontalToken
      tokenIndex={2}
      tokenName="Token B"
      tokenSymbol="TKB"
    />
  </div>
</div>
`;
